tap_github/repository_streams.py

```diff
@@ -1222,6 +1525,252 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
     ).to_dict()
 
 
+class PullRequestCommits(GitHubRestStream):
+    name = "pull_request_commits"
+    path = "/repos/{org}/{repo}/pulls/{pull_number}/commits"
+    ignore_parent_replication_key = True
+    primary_keys = ["node_id"]
+    parent_stream_type = PullRequestsStream
+    state_partitioning_keys = ["pull_number"]
```

Comment:
```suggestion
    state_partitioning_keys = ["org", "repo", "pull_number"]
```
You need all 3, or the state will be shared among all PRs with the same number :(

Reply:
@aaronsteers I've been wondering about this for `tap-gitlab` as well: with cases like this, the state object will become embarrassingly big, with one entry per PR essentially. In our case, we track all PRs over several hundred repos, this state object could end up having well into the ten of thousands of key/values if not more. As the tap emits one state object every N records, this ends up being very verbose (also thinking of memory consumption issues (see [this PR](https://github.com/oviohub/target-postgres/pull/2) for instance)).
Is there any best practice around managing such situations? I'm thinking there might be a way to only limit the state to "one level up", ie: `org, repo` only?

Reply:
Sounds good. I updated the partioning keys like you mentioned here: [0b3f52f](https://github.com/MeltanoLabs/tap-github/pull/93/commits/0b3f52fd4a44d5f999bbb054731a6f398abfd81b). I left in `project_id` with the logic that there aren't many projects per project. Let me know if you see any issues with this :+1: